nginx/setup.sh

```diff
@@ -1,82 +1,45 @@
-if [ -z "$1" ]; then
-	PROFILE=""
-else
-	PROFILE="--profile ${1}"
-fi
+CERT_DIRECTORY=$HOME/.gu/mkcert
+KEY_FILE=${CERT_DIRECTORY}/${DOMAIN}.key
+CERT_FILE=${CERT_DIRECTORY}/${DOMAIN}.crt
 
-aws ${PROFILE} s3 ls ${S3_BUCKET} 1>/dev/null 2>&1
-if [[ $? -gt 0 ]]; then
-	echo "ERROR: You do not have access to the Identity AWS account. Re-run with ${0} <profile name> to use a different profile."
-	exit 2
-fi
+# mkcert requires the JAVA_HOME envvar to be set to add the generated CA to Java's trust store
+# see https://github.com/FiloSottile/mkcert#supported-root-stores
+export JAVA_HOME=$(/usr/libexec/java_home)
 
-echo "Checking Nginx sites-enabled directory exists"
-ls "${nginxHome}/sites-enabled" 1>/dev/null 2>&1
-if [[ $? -gt 0 ]]; then
-	echo "ERROR: Missing ${nginxHome}/sites-enabled"
-    echo "Create sites-enabled directory and make sure it is included in your nginx.conf (usually in ${nginxHome}/nginx.conf):
-            http {
-                include       mime.types;
-                default_type  application/octet-stream;
-                # THIS IS WHAT YOU MUST ADD
-                include sites-enabled/*;
-            #..."
-	exit 1
-fi
+mkcert -install
 
-function install_ssh_certificate() {
-	KEY_NAME="${1}.key"
-	CRT_NAME="${1}.crt"
-	echo "Downloading SSL certificate $CRT_NAME from $S3_BUCKET"
-	aws ${PROFILE} s3 cp "${S3_BUCKET}${KEY_NAME}" ${DIR} 1>/dev/null
-	echo "Downloading SSL key $KEY_NAME from $S3_BUCKET"
-	aws ${PROFILE} s3 cp "${S3_BUCKET}${CRT_NAME}" ${DIR} 1>/dev/null
+echo -e "🔐 Creating certificate for: ${YELLOW}$@${NC}"
+mkdir -p ${CERT_DIRECTORY}
+mkcert -key-file=${KEY_FILE} -cert-file=${CERT_FILE} ${DOMAIN}
 
-	echo "Linking SSL certs/keys to $nginxHome"
-	sudo ln -fs "${DIR}/${CRT_NAME}" "${nginxHome}/${CRT_NAME}"
-	sudo ln -fs "${DIR}/${KEY_NAME}" "${nginxHome}/${KEY_NAME}"
-}
+echo -e "🔗 Symlinking the certificate for nginx at ${NGINX_HOME}"
+ln -sf ${KEY_FILE} ${NGINX_HOME}/${DOMAIN}.key
+ln -sf ${CERT_FILE} ${NGINX_HOME}/${DOMAIN}.crt
 
-function install_ssh_certificate_in_jdk_ca() {
-    echo "Importing SSL certificate into Java keystore ${jdkHome}/jre/lib/security/cacerts"
-	sudo "${jdkHome}/bin/keytool" -import -alias ${1} -keystore "${jdkHome}/jre/lib/security/cacerts" -file "${1}.crt" -storepass "changeit" -noprompt
-}
+echo -e "🔗 Symlinking nginx config file"
+ln -sf ${SOURCE_DIR}/frontend.conf ${NGINX_HOME}/servers/frontend.conf
 
-function install_nginx_configuration() {
-    echo "Linking Nginx configuration ${1} to $nginxHome/sites-enabled"
-    sudo ln -fs "$DIR/${1}" "$nginxHome/sites-enabled/${1}"
-}
-
-install_ssh_certificate ${SSL_CERT_NAME}
-install_ssh_certificate_in_jdk_ca ${SSL_CERT_NAME}
-install_nginx_configuration ${NGINX_SITE_CONF}
-
-echo "Restarting Nginx"
+echo -e "🚀 ${YELLOW}Restarting nginx, Requires sudo - enter password when prompted.${NC}"
 sudo nginx -s stop
 sudo nginx
 
-echo "Done. (To setup Dotcom Identity Frontend please follow identity-platform README.)"
```

Comment:
Presumably this requirement still exists for the `identity` project in dotcom to run on `https://profile.thegulocal.com`.  I'm not sure this message was ever really the best way to inform people of that requirement, but it would be good to reference it somehow (or ideally set it all up at once in `dev-nginx` or similar).

Reply:
From what I can tell, Identity have their own scripts in the platform repo? That is, this change shouldn't affect that process and we could/should update those scripts too?

Reply:
We should, but in order to run part of dotcom (identity) the identity-platform setup is currently a prerequisite - just wondering whether that is still clear from here?

Reply:
Ah! I'll add the message back.